x/emissions/keeper/inference_synthesis/synth_palette_weight.go

```diff
@@ -144,6 +160,15 @@ func (p *SynthPalette) CalcWeightedInference(weights RegretInformedWeights) (Inf
 		}
 	} else {
 		for _, inferer := range p.Inferers {
+			if _, ok := p.InferenceByWorker[inferer]; !ok {
+				continue
+			}
+			if _, ok := weights.inferers[inferer]; !ok {
+				continue
+			}
+			if _, ok := p.InfererRegrets[inferer]; !ok {
+				continue
+			}
```

Comment:
If we want to move forward with these checks, it might be a good idea to add some alerts here as well. Considering the actual implementation, If we ever fall into these check, we may have some critical issue in these calculations.

Reply:
adding debug logging just in case as suggested